api/management/README.md

```diff
@@ -0,0 +1,199 @@
+[OPA policy README] in services repo for details.
+
+Policies are automatically retrieved, based on the attestation scheme and
+tenant id for the evidence being evaluated. Currently, only one policy is
+allowed per tenant/scheme combination. If a policy is added for a tenant/scheme
+combination with an existing policy, the new policy replaces the old one. This
+also increments the internal policy version associated with the tenant/scheme
+combination.
+
+The policy management API exposes each scheme's policy via a unique resource
+path in the form `/policy/v1/{scheme}`. It supports the basic CRUD operations,
+typical to any resource management, in the following way:
+
+- **create**: A new policy is added to a scheme via a POST request to its path.
+- **read**: Policy can be retrieved via a GET request to its path.
+- **update**: A direct update is not possible, however a new policy can be
+              added and activated. Activating the new policy will deactivate
+              the old one, effectively replacing it.
+- **delete**: True deletion is not possible, in order to allow retrieval of the
+              specific policy version that was used to generate an attestation
+              result at a later date. However, all policies may be deactivated,
+              resulting in no policy being used during verification.
+
+[Attestation Results]: https://www.rfc-editor.org/rfc/internet-drafts/draft-fv-rats-ear-00.html
+[OPA Policy README]: https://github.com/veraison/services/blob/main/policy/README.opa.md#rules
+
+## REST API
+
+### Adding a new policy
+
+- A policy is added for a new scheme (with tenant implicit via auth).
+- Optionally, a name for the policy may be specified via query arguments. If a
+  name is not specified, the default name of `"default"` will be used.
+- On success, the new policy object is returned, with Location header set to
+  its unique URI.
+- The newly added policy is inactive. It must be activated via  a separate
+  request (see below).
+
+#### Request:
+```http
+POST /management/v1/policy/PSA_IOT&name=base HTTP/1.1
+Host: veraison.example
+Accept: application/vnd.veraison.policy+json
+
+hardware = "AFFIRMING" {
+    // secured state
+    state := bits.rsh(bits.and(evidence["psa-security-lifecycle"], 0xFFFF0000), 8)
+    state >= 0x3000
+    state <= 0x30FF
+} else = "CONTRAINDICATED"
+```
+
+#### Response:
+```http
+HTTP/1.1 201 Created
+Content-Type: application/vnd.veraison.policy+json
+Location: /policy/v1/PSA_IOT/340d22f7-9eda-499f-9aa2-5af295d6d812
+
+{
+  "type": "opa",
+  "name": "base",
+  "uuid": "340d22f7-9eda-499f-9aa2-5af295d6d812",
+  "active": false,
+  "ctime": "2023-07-06 00:00:00",
+  "rules": "hardware = \"AFFIRMING\" {\n          // secured state\n          state := bits.rsh(bits.and(evidence[\"psa-security-lifecycle\"], 0xFFFF0000), 8)\n          state >= 0x3000\n          state <= 0x30FF\n      } else = \"CONTRAINDICATED\"\n  }"
```

Comment:
this doesn't look much more readable than b64 to me :)
But then, I may have already transformed into a bot without knowing it...

Reply:
> this doesn't look much more readable than b64 to me :)

Seriosuly? You can tell at a glance that the hardware rule is being defined base on "psa-security-lifecycle"  from evidence.  Unless you can decode base64 in your head (in which cased, props!), this is demonstrably more readable.

Reply:
If I do:

```bash
jq -r '.rules | @base64d' < policy.json
```

I get a IMHO much better view of the content:
```rego
hardware = "AFFIRMING" {
    // secured state
    state := bits.rsh(bits.and(evidence["psa-security-lifecycle"], 0xFFFF0000), 8)
    state >= 0x3000
    state <= 0x30FF
} else = "CONTRAINDICATED"
```
than
```
"hardware = \"AFFIRMING\" {\n    // secured state\n    state := bits.rsh(bits.and(evidence[\"psa-security-lifecycle\"], 0xFFFF0000), 8)\n    state >= 0x3000\n    state <= 0x30FF\n} else = \"CONTRAINDICATED\"\n"
```
and since I am assuming the interaction pattern we are targeting here is that of an admin with a shell, `jq(1)` looks like a not so unreasonable expectation.

Reply:
Yes,  the problem is you have to do that -- conversely, this way you can immediately read the policy rules without any additionaly processing if you need to quickly verify something (e.g. that  a particular rule is present).  This is a lot more convienent, e..g, if you're  dumping the response inside a debugger.

Reply:
Note that you have the ability to post-porcess, you can get an identifical output:
```bash
$ eval "echo -e $(jq '.rules' < /tmp/policy.json)"
hardware = "AFFIRMING" {
          // secured state
          state := bits.rsh(bits.and(evidence["psa-security-lifecycle"], 0xFFFF0000), 8)
          state >= 0x3000
          state <= 0x30FF
      } else = "CONTRAINDICATED"
  }
```

Reply:
If we're talking about structured logging with associated tooling, robustness inot an issue, as they will handle escaping correctly. If we're talking about unstructured logging that is parsed with ad-hoc scripts, they you are right -- they scripts will need to handle escpaing correctly, and it a lot easier and safer to deal with basey64. However that's a problem with scripting over undstructured logs. For me, ease of debugging trumps that.